libraries/chain/transaction_context.cpp

```diff
@@ -296,9 +300,15 @@ namespace eosio { namespace chain {
                           "not enough time left in block to complete executing transaction",
                           ("now", now)("deadline", _deadline)("start", start)("billing_timer", now - pseudo_start) );
             } else if( deadline_exception_code == tx_cpu_usage_exceeded::code_value ) {
-               EOS_THROW( tx_cpu_usage_exceeded,
-                          "transaction was executing for too long",
-                          ("now", now)("deadline", _deadline)("start", start)("billing_timer", now - pseudo_start) );
+               if (cpu_limit_due_to_greylist) {
+                  EOS_THROW( greylist_cpu_usage_exceeded,
+                           "transaction was executing for too long",
+                           ("now", now)("deadline", _deadline)("start", start)("billing_timer", now - pseudo_start) );
+               } else {
+                  EOS_THROW( tx_cpu_usage_exceeded,
+                           "transaction was executing for too long",
```

Comment:
Would like to see a diff string for these two cases, to make it more obvious if grepping.